tests/actor/test_actor_reentrancy.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+
+"""
+Copyright (c) Microsoft Corporation and Dapr Contributors.
+Licensed under the MIT License.
+"""
+
+import unittest
+import asyncio
+
+from unittest import mock
+
+from dapr.actor.runtime.runtime import ActorRuntime
+from dapr.actor.runtime.config import ActorRuntimeConfig, ActorReentrancyConfig
+from dapr.serializers import DefaultJSONSerializer
+
+from tests.actor.fake_actor_classes import (
+    FakeReentrantActor,
+    FakeMultiInterfacesActor,
+    FakeSlowReentrantActor,
+)
+
+from tests.actor.utils import _run
+
+
+class ActorRuntimeTests(unittest.TestCase):
+    def setUp(self):
+        ActorRuntime._actor_managers = {}
+        ActorRuntime.set_actor_config(
+            ActorRuntimeConfig(reentrancy=ActorReentrancyConfig(enabled=True)))
+        self._serializer = DefaultJSONSerializer()
+        _run(ActorRuntime.register_actor(FakeReentrantActor))
+        _run(ActorRuntime.register_actor(FakeSlowReentrantActor))
+        _run(ActorRuntime.register_actor(FakeMultiInterfacesActor))
+
+    def test_reentrant_dispatch(self):
+        _run(ActorRuntime.register_actor(FakeMultiInterfacesActor))
+
+        request_body = {
+            "message": "hello dapr",
+        }
+
+        reentrancy_id = "0faa4c8b-f53a-4dff-9a9d-c50205035085"
+
+        test_request_body = self._serializer.serialize(request_body)
+        response = _run(ActorRuntime.dispatch(
+            FakeMultiInterfacesActor.__name__, 'test-id',
+            "ReentrantMethod", test_request_body, reentrancy_id=reentrancy_id))
```

Comment:
To confirm, this is just validating that we can call a reentrant actor normally right? Not actually testing the reentrant portion?

Reply:
This simulates that a reentrant can be called normally way (as the final call in a reentrancy chain for example) but it only tests the internal part where the HTTP library has already parsed the reentrancy header and then calls the runtime dispatch function shown here. This particular test doesn't test calling a reentrant actor from a reentrant actor. That's done in another test.